### PR TITLE
COMMERCE-9364, COMMERCE-9365 | master

### PR DESCRIPTION
**Pull request by AfonsoHSantos**
https://issues.liferay.com/browse/COMMERCE-9364
https://issues.liferay.com/browse/COMMERCE-9365

### CODE REVIEW

**Comment by AfonsoHSantos** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAddressPermissions.testcase [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+definition {
+
+	property custom.properties = "jsonws.web.service.paths.excludes=${line.separator}company.security.strangers.verify=false";
+	property portal.release = "true";
+	property portal.upstream = "true";
+	property release.feature.flags.disable.DISABLE_PRIVATE_LAYOUTS = "true";
+	property testray.main.component.name = "Order Management";
+
+	setUp {
+		CommerceConfiguration.commerceSetUp();
```
Comment (Kvale1733) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAddressPermissions.testcase]: Change to CommerceConfiguration.commerceSetUp(minium = "true"); to create Minium site in setup.

**Comment by Kvale1733** [comment]
You can combine both tests into one. The billing step is right after the shipping step.

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAddressPermissions.testcase [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+			CommerceLogin.miniumLogin(
+				password = "test",
+				urlAppend = "web/minium",
+				userEmailAddress = "buyer@liferay.com");
+
+            FrontStore.addInitializerProductToCart(
+                productName = "ABS Sensor",
+                productQuantity = "1");
+		}
+	}
```
Comment (Kvale1733) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAddressPermissions.testcase]: These steps should be moved into the test. We might have other tests later on that use different setup.

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAddressPermissions.testcase [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+				key_navItem = "Accounts",
+				key_subItem = "Accounts");
+				
+			Uncheck.uncheckNotVisible(
+				locator1 = "RolesPermissionsNavigation#PERMISSIONS_ITEM_CHECKBOX",
+				key_itemName = "Account Entry",
+				key_checkboxName = "MANAGE_ADDRESSES");
+
+			Button.clickSave();
+		}
```
Comment (Kvale1733) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAddressPermissions.testcase]: You can replace all these steps with one json macro:

Permissions.removePermissionViaAPI(
				groupName = "Minium",
				resourceAction = "MANAGE_ADDRESSES",
				resourceName = "com.liferay.account.model.AccountEntry",
				roleTitle = "Buyer",
				roleType = "account");

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAddressPermissions.testcase [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+	}
+
+    @description = "COMMERCE-9364 | Without Manage Address permission, user cannot add new shipping address and only can select pre-defined addresses"
+	@priority = "3"
+	test CanOnlySelectPreDefinedShippingAddressWhithouPermission {
+		task ("When the buyer reach the Billing Address step") {
+			CommerceCheckout.gotoCheckout();
+		}
+
+		task ("Then the buyer can select a billing address between the pre-defined list of existing billing addresses") {
```
Comment (Kvale1733) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAddressPermissions.testcase]: The task should actually be for shipping address not billing addresses

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAddressPermissions.testcase [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+				locator1 = "Select#ANY",
+				value1 = "Liferay Billing and Shipping");
+
+			Select(
+				key_selectFieldLabel = "Choose Shipping Address",
+				locator1 = "Select#ANY",
+				value1 = "Liferay Billing and Shipping");
+		}
+
+		task ("And then the 'Use shipping address as billing address' checkbox is not visible") {
```
Comment (Kvale1733) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAddressPermissions.testcase]: The tasks can just be "And the...." No need for "And then"

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAddressPermissions.testcase [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+			AssertElementNotPresent(
+				locator1 = "Checkbox#ANY_CHECKBOX",
+				checkboxName = "Use shipping address as billing address");
+		}
+
+		task ("But the buyer is not able to add a new shipping address as fields are disabled") {
```
Comment (Kvale1733) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAddressPermissions.testcase]: Same here. Change the tasks to "And the buyer.."

**Comment by AfonsoHSantos** [comment]
ci:test:sf

**Comment by Kvale1733** [comment]
testing here: https://github.com/Kvale1733/liferay-portal/pull/113

**Comment by Kvale1733** [comment]
ci:close

**Pull request closed without merging**